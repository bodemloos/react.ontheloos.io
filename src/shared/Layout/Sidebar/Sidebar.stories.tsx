import type { Meta, StoryObj } from "@storybook/react";

import { BrowserRouter } from "shared/Router";

import { Sidebar } from "./index";

const meta = {
  component: Sidebar,
  decorators: [
    (Story) => (
      <BrowserRouter>
        <div style={{ height: "100vh", width: "250px" }}>
          <Story />
        </div>
      </BrowserRouter>
    ),
  ],
  parameters: {
    layout: "left",
  },
} satisfies Meta<typeof Sidebar>;

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    title: "login to see nav. items",
  },
};
