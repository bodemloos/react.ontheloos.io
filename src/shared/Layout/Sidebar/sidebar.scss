.sidebar::before {
  content: "";
  display: block;
  background: url(/sidebar-background.jpeg) center 0px no-repeat;
  height: 100%;
  position: absolute;
  top: 0;
  left: 0;
  width: 250px;
  filter: opacity(0.08);
  background-repeat: no-repeat;
  background-size: cover;
}
