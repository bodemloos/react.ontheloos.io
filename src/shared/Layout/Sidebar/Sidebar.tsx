import { Box, Flex, Icon as ThemeIcon } from "@chakra-ui/react";

import { Separator } from "shared/Separator";

import { INavItem } from "../Navbar/INav";
import { useNavItems } from "../Navbar/useNavItems";
import { SidebarItem } from "./SidebarItem";
import "./sidebar.scss";

export interface ISubbarProps {
  title?: string;
}

export const Sidebar: React.FC = ({ title = "DASHBOARD" }: ISubbarProps) => {
  const navItems = useNavItems();
  return (
    <Box pt={4} h="full" className="sidebar">
      <Flex direction="column" h="full">
        <Box>
          <Box pb={5} pt={2} pr="10px" textAlign="center">
            {title}
          </Box>
          <Separator mb={4} marginLeft="-10px" w={"100%"} />
          <Box as="nav" px={0}>
            {navItems.map((nav) => (
              <Box key={nav.label}>
                <Flex
                  color="gray.200"
                  bg="gray.700"
                  w="65%"
                  borderTopRightRadius={10}
                  borderBottomRightRadius={10}
                  fontSize={12}
                  px={3}
                  py={4}
                  mb={3}
                  mt={3}
                >
                  <ThemeIcon as={nav.icon} />
                  <Box lineHeight={1} pl={5}>
                    {nav.label}
                  </Box>
                </Flex>
                {nav.children.map((navItem: INavItem) => (
                  <SidebarItem
                    key={navItem.label}
                    href={navItem.href || ""}
                    label={navItem.label}
                    icon={navItem.icon}
                  />
                ))}
              </Box>
            ))}
          </Box>
        </Box>
      </Flex>
    </Box>
  );
};
