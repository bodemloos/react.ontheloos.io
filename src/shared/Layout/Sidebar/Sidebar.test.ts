import renderer from "react-test-renderer";

import { Sidebar } from "./Sidebar";

test("<Sidebar /> renders", () => {
  const tree = renderer.create(Sidebar).toJSON();

  expect(tree).toMatchSnapshot();
});
