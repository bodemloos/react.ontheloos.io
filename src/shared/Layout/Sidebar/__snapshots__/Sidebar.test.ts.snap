// Vitest Snapshot v1

exports[`<Sidebar /> renders 1`] = `null`;
